jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialWriter.kt

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataKeys
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.vfs.LocalFileSystem
+import software.amazon.awssdk.profiles.ProfileFileSystemSetting
+import software.amazon.awssdk.utils.JavaSystemSetting
+import software.amazon.awssdk.utils.StringUtils
+import software.aws.toolkits.resources.message
+import java.io.File
+import java.nio.file.FileSystems
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+class CreateOrUpdateCredentialProfilesAction : AnAction(message("configure.toolkit.upsert_credentials.action")), DumbAware {
+    private val credentialCreator = CreateOrUpdateCredentialProfiles.getInstance()
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.getRequiredData(DataKeys.PROJECT)
+        credentialCreator(project) {
+            Messages.showOkCancelDialog(
```

Comment:
Messages has a back door to set an expected yes/no, if you want to simplify this.
`Messages.setTestDialog(TestDialog.OK)`

Reply:
Oooh - awesome - good call. Much simplified.